VirtoCommerce.Storefront/Controllers/AccountController.cs

```diff
@@ -301,101 +301,100 @@ public async Task<ActionResult> Login([FromForm] Login login, [FromQuery]string
+                                Recipient = GetUserEmail(user)
+                            };
+                        }
+                        var sendingResult = await SendNotificationAsync(twoFactorNotification);
+
+                        if (sendingResult.IsSuccess != true)
+                        {
+                            WorkContext.Form.Errors.Add(SecurityErrorDescriber.ErrorSendNotification(sendingResult.ErrorMessage));
+                            return View("customers/login", WorkContext);
+                        }
+
+                        WorkContext.Form = Form.FromObject(veryfyCodeViewModel);
+
+                        return View("customers/verify_code", WorkContext);
                     }
 
-                    twoFactorNotification = new TwoFactorSmsNotification(WorkContext.CurrentStore.Id, WorkContext.CurrentLanguage)
+                    if (loginResult.IsLockedOut)
                     {
-                        Token = code,
-                        Recipient = phoneNumber,
-                    };
+                        return View("lockedout", WorkContext);
+                    }
 
-                }
-                else // "Email"
-                {
-                    twoFactorNotification = new TwoFactorEmailNotification(WorkContext.CurrentStore.Id, WorkContext.CurrentLanguage)
+                    if (loginResult is CustomSignInResult signInResult && signInResult.IsRejected)
                     {
-                        Token = code,
-                        Sender = WorkContext.CurrentStore.Email,
-                        Recipient = GetUserEmail(user)
-                    };
-                }
-                var sendingResult = await SendNotificationAsync(twoFactorNotification);
+                        WorkContext.Form.Errors.Add(SecurityErrorDescriber.AccountIsBlocked());
+                    }
 
-                if (sendingResult.IsSuccess != true)
+                }
+                else
                 {
-                    WorkContext.Form.Errors.Add(SecurityErrorDescriber.ErrorSendNotification(sendingResult.ErrorMessage));
-                    return View("customers/login", WorkContext);
+                    WorkContext.Form.Errors.Add(SecurityErrorDescriber.UserCannotLoginInStore());
                 }
 
-                WorkContext.Form = Form.FromObject(veryfyCodeViewModel);
-
-                return View("customers/verify_code", WorkContext);
+                WorkContext.Form.Errors.Add(SecurityErrorDescriber.LoginFailed());
             }
-
-            if (loginResult.IsLockedOut)
-            {
-                return View("lockedout", WorkContext);
-            }
-
-            if (loginResult is CustomSignInResult signInResult && signInResult.IsRejected)
+            else
             {
-                WorkContext.Form.Errors.Add(SecurityErrorDescriber.AccountIsBlocked());
+                WorkContext.Form.Errors.Add(SecurityErrorDescriber.OperationFailed());
```

Comment:
Since we do not follow the single return rule in this controller, to improve readability please make handling of 
```
if (user == null) 
{
…
}
```
right after getting the user.

Reply:
Let's return `SecurityErrorDescriber.LoginFailed()` instead of `SecurityErrorDescriber.OperationFailed()` in case of non-existent user, to unify wrong password result woth non existent user.

Reply:
Done